src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -2754,10 +2756,66 @@ impl Coordinator {
 
     fn explain_materialized_view(
         &mut self,
-        _ctx: &mut ExecuteContext,
-        _plan: plan::ExplainPlan,
+        ctx: &mut ExecuteContext,
+        plan: plan::ExplainPlan,
     ) -> Result<ExecuteResponse, AdapterError> {
-        coord_bail!("unsupported operation: explain_materialized_view");
+        let plan::ExplainPlan {
+            raw_plan: _,
+            row_set_finishing,
+            stage,
+            format,
+            config,
+            no_errors: _,
+            explainee,
+        } = plan;
+
+        assert!(row_set_finishing.is_none());
+
+        let Explainee::MaterializedView(id) = explainee else {
+            unreachable!()
+        };
+
+        if !matches!(stage, ExplainStage::OptimizedPlan) {
+            coord_bail!(
+                "cannot EXPLAIN {} FOR MATERIALIZED VIEW\n\nonly {} is supported",
+                stage,
+                ExplainStage::OptimizedPlan
+            );
+        }
+
+        let CatalogItem::MaterializedView(_) = self.catalog().get_entry(&id).item() else {
+            // This is currently asserted in the planner.
+            // However, this assumtion might change when we make changes for #18089.
+            unreachable!()
+        };
+
+        let Some(mut optimized_plan) = self.catalog().try_get_optimized_plan(&id).cloned() else {
+            coord_bail!("cannot find OPTIMIZED PLAN FOR MATERIALIZED VIEW in catalog");
```

Comment:
We expect to always have a plan here right? In that case, add a soft_assert and `trace::error` which will 1) panic during dev and 2) log to sentry and not panic in prod.

Reply:
> 2) log to sentry and not panic in prod.

The problem is if the `optimized_plan` is missing I cannot recover and continue execution on the happy path, so I cannot just replace `coord_bail!` with `soft_panic_or_log!` for example.